src/GatewayMethod.php

```diff
@@ -764,18 +753,14 @@ public function returnProcess($request, RedirectInformation $transactionInfo, $i
                     );
                 }
 
-                if ($status == $sessionStatusInstance::ST_APPROVED) {
-                    $this->msg['message'] = $this->msg_approved;
-                    $this->msg['class'] = 'woocommerce-message';
-
-                    if ($paymentStatus !== Status::ST_APPROVED) {
-                        $payment = $transactionInfo->lastApprovedTransaction();
+                if ($status == $sessionStatusInstance::ST_APPROVED && $paymentStatus !== Status::ST_APPROVED) {
+                    $payment = $transactionInfo->lastApprovedTransaction();
+                    $this->logger(json_encode($payment));
 
-                        $order->add_order_note($this->getOrderNote($order->get_id(), $payment, $status, $totalAmount));
-                        $order->add_meta_data('placetopay_response', json_encode($payment->toArray()));
-                        $order->payment_complete();
-                        $this->logger('Payment approved for order # ' . $order->get_id(), __METHOD__);
-                    }
+                    $order->add_order_note($this->getOrderNote($order->get_id(), $payment, $status, $totalAmount));
+                    $order->add_meta_data('placetopay_response', json_encode($payment->toArray()));
+                    $order->payment_complete();
+                    $this->logger('Payment approved for order # ' . $order->get_id(), __METHOD__);
                 } else {
```

Comment:
Review behavior when the first transaction is rejected but the second is: approved or approved_partial